### PR TITLE
Improved `selectOptions` documentation and typescript definition

### PR DESCRIPTION
**Pull request by swistach**
### Context
Accordingly to the reported mismatch (#8488) between what that option is doing and how we described it in our documentation - I improved JSDocs and d.ts file to cover supported use cases.

Additionally, this PR contains new tests for TS definition and missing the E2E test for using `selectOptions` as a function.

### How has this been tested?
Used each supported type for `selectOptions` option.
- `string[]`
- `{ [prop: string]: string }`
- `(visualRow: number, visualColumn: number, prop: string | number) => string[] | { [prop: string]: string }`

### Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. #8488 

### Affected project(s):
- [x] `handsontable`

### CODE REVIEW

**Review by wszymanski — DISMISSED** [review]
LGFM 🎉

File: .changelogs/8488.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "title": "Improved documentation and TypeScript definition for 'selectOptions' option.",
```
Comment (kirszenbaum) [inline .changelogs/8488.json:2]:
```suggestion
  "title": "Improved the documentation and TypeScript definition of the `selectOptions` option.",
```

File: src/dataMap/metaManager/metaSchema.js [hunk 2]
```diff
@@ -2382,8 +2382,26 @@ export default () => {
      * columns: [
      *   {
      *     editor: 'select',
-     *     // add three select options to choose from
+     *     // as an array of strings, `option.value` and `option.textContent` uses the same value
```
Comment (kirszenbaum) [inline src/dataMap/metaManager/metaSchema.js:2385]:
```suggestion
     *     // as an array of strings: `option.value` and `option.textContent` use the same value
```

File: src/dataMap/metaManager/metaSchema.js [hunk 3]
```diff
@@ -2382,8 +2382,26 @@ export default () => {
...
      *     selectOptions: ['A', 'B', 'C'],
+     *     // as an object - `option.value` points a key, `option.textContent` contains string assigned to the key
```
Comment (kirszenbaum) [inline src/dataMap/metaManager/metaSchema.js:2387]:
```suggestion
     *     // as an object: `option.value` appoints a key, and `option.textContent` contains a string assigned to the key
```
```suggestion
     *     // as an object - `option.value` points a key, `option.textContent` contains string assigned to the key
```

File: src/dataMap/metaManager/metaSchema.js [hunk 4]
```diff
@@ -2382,8 +2382,26 @@ export default () => {
...
+     *     selectOptions: {
+     *       value1: 'Label 1',
+     *       value2: 'Label 2',
+     *       value3: 'Label 3',
+     *     },
+     *     // as a function returning possible options as an array
```
Comment (kirszenbaum) [inline src/dataMap/metaManager/metaSchema.js:2393]:
```suggestion
     *     // as a function that returns possible options as an array
```

File: src/dataMap/metaManager/metaSchema.js [hunk 5]
```diff
@@ -2382,8 +2382,26 @@ export default () => {
...
+     *     selectOptions(visualRow, visualColumn, prop) {
+     *       return ['A', 'B', 'C'];
+     *     },
+     *     // as a function returning possible options as an object
```
Comment (kirszenbaum) [inline src/dataMap/metaManager/metaSchema.js:2397]:
```suggestion
     *     // as a function that returns possible options as an object
```

**Review by kirszenbaum — CHANGES_REQUESTED** [review]

**Review by wszymanski — APPROVED** [review]

**Review by kirszenbaum — APPROVED** [review]

**Pull request merged**